pages/code_schools.js

```diff
@@ -38,278 +39,268 @@ const moocSchools = [
-    errorMessage: '',
-  };
+CodeSchools.defaultProps = {
+  allSchools: [],
+  errorMessage: '',
+};
 
-  /* eslint-disable react/destructuring-assignment */
-  state = {
-    filteredSchools: this.props.allSchools,
-    selectedStates: [],
-    locationsModalInfo: { name: '', locations: [] },
-  };
-  /* eslint-enable react/destructuring-assignment */
+function CodeSchools({ allSchools, errorMessage }) {
+  const [filteredSchools, setFilterSchools] = useState(allSchools);
+  const [selectedStates, setSelectedStates] = useState([]);
+  const [locationsModalInfo, setLocationsModalInfo] = useState({ name: '', locations: [] });
 
-  handleModalOpen = ({ name, locations }) =>
-    this.setState({ locationsModalInfo: { name, locations } });
+  const handleModalOpen = ({ name, locations }) => setLocationsModalInfo({ name, locations });
 
-  handleModalClose = () => this.setState({ locationsModalInfo: { name: '', locations: [] } });
+  const handleModalClose = () => setLocationsModalInfo({ name: '', locations: [] });
 
-  filterOnline = () => {
-    const { allSchools } = this.props;
+  const filterOnline = () => {
     const onlineSchools = allSchools.filter(school => school.hasOnline);
-
-    this.setState({ filteredSchools: onlineSchools, selectedStates: [] });
+    setFilterSchools(onlineSchools);
+    setSelectedStates([]);
   };
 
-  filterState = selectedOptions => {
+  const filterState = selectedOptions => {
     if (!selectedOptions) {
-      this.showAllSchools();
+      showAllSchools();
       return;
     }
-    const { allSchools } = this.props;
     const states = selectedOptions.map(state => state.value);
     const stateSchools = allSchools.filter(school =>
       school.locations.some(location => states.includes(location.state)),
     );
 
     if (states.length > 0) {
-      this.setState({ filteredSchools: stateSchools, selectedStates: selectedOptions });
+      setFilterSchools(stateSchools);
+      setSelectedStates(selectedOptions);
     } else {
-      this.showAllSchools();
+      showAllSchools();
     }
   };
 
-  filterVaApproved = () => {
-    const { allSchools } = this.props;
+  const filterVaApproved = () => {
     const vaApproved = allSchools.filter(school =>
       school.locations.some(location => location.vaAccepted),
     );
-    this.setState({ filteredSchools: vaApproved, selectedStates: [] });
+    setFilterSchools(vaApproved);
+    setSelectedStates([]);
   };
 
-  filterVetTecApproved = () => {
-    const { allSchools } = this.props;
+  const filterVetTecApproved = () => {
     const vetTecApprovedSchools = allSchools.filter(school => school.isVetTecApproved);
-    this.setState({ filteredSchools: vetTecApprovedSchools, selectedStates: [] });
+    setFilterSchools(vetTecApprovedSchools);
+    setSelectedStates([]);
   };
 
-  showAllSchools = () => {
-    const { allSchools } = this.props;
-    this.setState({ filteredSchools: allSchools, selectedStates: [] });
+  const showAllSchools = () => {
```

Comment:
Can you define this near the top of the method so calls are below the definition?

Reply:
Yup, no problem. :+1: